adobe_vipm/adobe/config.py

```diff
@@ -148,65 +172,78 @@ def get_preferred_language(self, country: str) -> str:
+            return json.load(config_file)
 
     def _setup(self):
-        config_data = self._load_config()
-        credentials_data = self._load_credentials()
         authorizations_data = self._load_authorizations()
+        credentials_map = map_by("authorization_uk", self._load_credentials())
 
-        credentials_map = {cred["authorization_uk"]: cred for cred in credentials_data}
         for authorization_data in authorizations_data["authorizations"]:
-            auth_uk = authorization_data["authorization_uk"]
-            authorization = Authorization(
-                authorization_uk=auth_uk,
-                authorization_id=authorization_data.get("authorization_id"),
-                name=credentials_map[auth_uk]["name"],
-                client_id=credentials_map[auth_uk]["client_id"],
-                client_secret=credentials_map[auth_uk]["client_secret"],
-                currency=authorization_data["currency"],
-                distributor_id=authorization_data["distributor_id"],
-            )
-            self.authorizations[auth_uk] = authorization
+            authorization = self._create_authorization(authorization_data, credentials_map)
+            self.authorizations[authorization_data["authorization_uk"]] = authorization
 
             if authorization.authorization_id:  # pragma: no branch
                 self.authorizations[authorization.authorization_id] = authorization
 
-            for reseller_data in authorization_data["resellers"]:
-                seller_uk = reseller_data["seller_uk"]
-                seller_id = reseller_data.get("seller_id")
-                reseller = Reseller(
-                    id=reseller_data["id"],
-                    seller_uk=seller_uk,
-                    authorization=authorization,
-                    seller_id=seller_id,
-                )
-                self.resellers[(authorization, seller_uk)] = reseller
-
-                if seller_id:  # pragma: no branch
-                    self.resellers[(authorization, seller_id)] = reseller
-
-        self.language_codes = config_data["language_codes"]
-        for country in config_data["countries"]:
+            self._setup_resellers(authorization, authorization_data)
+
+        self._setup_countries(self._load_config())
+
+    def _setup_resellers(self, authorization: Authorization, authorization_data: dict):
+        for reseller_data in authorization_data["resellers"]:
+            seller_uk = reseller_data["seller_uk"]
+            seller_id = reseller_data.get("seller_id")
+            reseller = Reseller(
+                id=reseller_data["id"],
+                seller_uk=seller_uk,
+                authorization=authorization,
+                seller_id=seller_id,
+            )
+            self.resellers[authorization, seller_uk] = reseller
+
+            if seller_id:  # pragma: no branch
+                self.resellers[authorization, seller_id] = reseller
```

Comment:
```suggestion
            seller_id = seller_id if seller_id else seller_uk
            self.resellers[authorization, seller_id] = reseller
```

Reply:
That's not true, it setup values to both keys, seller_id and seller_uk. For seller id only if it exists